src/ir.cpp

```diff
@@ -9771,13 +9771,41 @@ IrInstruction *ir_eval_const_value(CodeGen *codegen, Scope *scope, AstNode *node
     return ir_exec_const_result(codegen, analyzed_executable);
 }
 
-static ZigType *ir_resolve_type(IrAnalyze *ira, IrInstruction *type_value) {
+static ZigType *ir_resolve_type(IrAnalyze *ira, IrInstruction *type_value, ZigTypeId wanted_type) {
     if (type_is_invalid(type_value->value.type))
         return ira->codegen->builtin_types.entry_invalid;
 
+    const char *expected_type_str = "'type'";
+
+    switch (wanted_type) {
+        case ZigTypeIdEnum:
+            expected_type_str = "enum";
+            break;
+        case ZigTypeIdErrorSet:
+            expected_type_str = "error set";
+            break;
+        case ZigTypeIdFn:
+            expected_type_str = "function";
+            break;
+        case ZigTypeIdInt:
+            expected_type_str = "integer";
+            break;
+        case ZigTypeIdPointer:
+            expected_type_str = "pointer";
+            break;
+        case ZigTypeIdStruct:
+            expected_type_str = "struct";
+            break;
+        case ZigTypeIdInvalid:
+            break;
+        default:
+            expected_type_str = type_id_name(wanted_type);
+            return ira->codegen->builtin_types.entry_invalid;
```

Comment:
I was only replacing code where the formula matched:

```cc
ZigType *payload_type = ir_resolve_type(ira, instruction->payload->child);
if (type_is_invalid(payload_type))
    return ira->codegen->invalid_instruction;
```
followed by:

```cc
if (err_set_type->id != ZigTypeIdErrorSet) {
    ir_add_error(ira, instruction->err_set->child,
        buf_sprintf("expected error set type, found type '%s'",
            buf_ptr(&err_set_type->name)));
    return ira->codegen->invalid_instruction;
}
```

so that the error messages would be identical.

I do like mapping to `type_id_name`, so I will revamp the code likewise.